### PR TITLE
Integration tests: run git daemon on a random-but-bind()able port

### PR DESCRIPTION
**Pull request by nalind**
#### What type of PR is this?

/kind flake

#### What this PR does / why we need it:

Use a listener helper to bind to an available-according-to-the-kernel listening port and run a command with its stdio more or less tied to the connection instead of trying to launch a git daemon directly using a port number that we can only guess is available.

#### How to verify it

Our build-using-a-git-context tests should continue to pass.

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by Luap99 — COMMENTED** [review]
This looks like a bit overkill, in podman we have this long solved by using random_free_port() which seem to hold up fine there.
https://github.com/containers/podman/blob/740f1d1fc710d4560d2463b11333cc9ad68e22bd/test/system/helpers.network.bash#L306-L329

I think this is simpler for test issues to debug then dealing with custom written binary that can fail in a lot of unexpected ways. It will also means we would have to ship this binary in the test package for gating tests

**Comment by Luap99** [comment]
Well debugging this custom go code may be easier for you but I doubt it is easier for the person running these tests on gating env that have to look at any failures and make sense of them such as @edsantiago.

In any case you are the main maintainer here and I am certainly not going to block this if it helps to solve a flake.

File: tests/inet/inet.go
Comment (Luap99) [inline tests/inet/inet.go:34]: should this exit(1)?

File: tests/inet/inet.go
Comment (Luap99) [inline tests/inet/inet.go:37]: would it make sense to at least bind 127.0.0.1, there seems little reason to expose this externally

File: tests/inet/inet.go
Comment (Luap99) [inline tests/inet/inet.go:49]: you can use `net.SplitHostPort()`

File: tests/inet/inet.go
Comment (nalind) [reply tests/inet/inet.go:34]: Yeah, probably.  Changing it.

File: tests/inet/inet.go
Comment (nalind) [reply tests/inet/inet.go]: Oh, nice.  Switching to it.

File: tests/inet/inet.go
Comment (nalind) [reply tests/inet/inet.go:36]: Hmm, leaving it empty means we don't have to care about ipv4/ipv6 here, though.

File: rpm/buildah.spec
Comment (Luap99) [inline rpm/buildah.spec:136]: this also needs to be added to the install section like `bin/copy` and then listed under `%files tests`

And since we had this discussion last week about podman-tests do we ship buildah-tests in RHEL as well? I really doubt we want to support these test helpers?

File: tests/inet/inet.go
Comment (Luap99) [reply tests/inet/inet.go:36]: I mean fair enough, it should not be a problem though as all test connect to localhost which generally should try both `::1` and `127.0.0.1` so I doubt it causes issues.

**Review by Luap99 — COMMENTED** [review]
go code LGTM

File: rpm/buildah.spec
Comment (nalind) [reply rpm/buildah.spec:136]: Ah, right.  Adding.

File: Makefile
Comment (nalind) [reply Makefile:113]: The scripts will need to be told where it is regardless, since it won't be in `../bin` relative to where the scripts end up packaged.  At that point it's the packager's decision, as it is with the other test helpers.

**Review by Luap99 — APPROVED** [review]
LGTM

**Pull request merged**
